### PR TITLE
Using Get-CmiInstance to detect Hyper-V availability

### LINKED ISSUE #13571: Use `Get-CimInstance` to detect Hyper-V availability
### What Happened?

https://github.com/kubernetes/minikube/blob/9f96701d8f751d5dfce72f2fb762833f4c396747/pkg/minikube/registry/drvs/hyperv/hyperv.go#L94

[`Get-WmiObject`](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.management/get-wmiobject?view=powershell-5.1) is deprecated and has been superseded by [`Get-CimInstance`](https://docs.microsoft.com/en-us/powershell/module/cimcmdlets/get-ciminstance?view=powershell-7.2) in 2012 with the release of PowerShell 3.0. Replacing it should be easy and allow to run minikube on systems where the deprecated WMI cmdlets have been blocked by policy.

### Attach the log file

-

### Operating System

Windows

### Driver

Hyper-V

### PR DESCRIPTION
**Pull request by NikhilSharmaWe**
**Replacing deprecated Get-Wmiobject with Get-CimInstance to detect Hyper-V availability.**
Fixes #13571

### CODE REVIEW

**Comment by medyagh** [comment]
@spowelljr review plz

File: pkg/minikube/registry/drvs/hyperv/hyperv.go [hunk 1]
```diff
@@ -91,7 +91,7 @@ func status() registry.State {
 	ctx, cancel := context.WithTimeout(context.Background(), 8*time.Second)
 	defer cancel()
 
-	cmd := exec.CommandContext(ctx, path, "-NoProfile", "-NonInteractive", "@(Get-Wmiobject Win32_ComputerSystem).HypervisorPresent")
+	cmd := exec.CommandContext(ctx, path, "-NoProfile", "-NonInteractive", "@(Get-CimInstance Win32_ComputerSystem).HypervisorPresent")
```
Comment (medyagh) [inline pkg/minikube/registry/drvs/hyperv/hyperv.go:94]: will this work on Old windows versions ?

how about we put the old way  as Fail Over option ? if this fails lets try the old way (maybe someone with old windows) ?

Comment (NikhilSharmaWe) [reply pkg/minikube/registry/drvs/hyperv/hyperv.go:94]: I think that will be a nice addition
Should add it and you can review the changes for improvement ?

File: pkg/minikube/registry/drvs/hyperv/hyperv.go [hunk 2]
```diff
@@ -91,13 +91,19 @@ func status() registry.State {
...
 	if err != nil {
-		errorMessage := fmt.Errorf("%s failed:\n%s", strings.Join(cmd.Args, " "), out)
-		fixMessage := "Start PowerShell as an Administrator"
-		return registry.State{Installed: false, Running: true, Error: errorMessage, Fix: fixMessage, Doc: docURL}
+		errorMessage1 := fmt.Errorf("%s failed:\n%s ", strings.Join(cmd.Args, " "), out)
+		cmd := exec.CommandContext(ctx, path, "-NoProfile", "-NonInteractive", "@(Get-Wmiobject Win32_ComputerSystem).HypervisorPresent")
+		out, err := cmd.CombinedOutput()
+		if err != nil {
+			errorMessage2 := fmt.Errorf("%s failed:\n%s ", strings.Join(cmd.Args, " "), out)
+			errorMessage := fmt.Errorf("%s\n%s", errorMessage1, errorMessage2)
```
Comment (spowelljr) [inline pkg/minikube/registry/drvs/hyperv/hyperv.go:103]: Can we improve the variable names of the error messages, we have `errorMessage`, `errorMessage1`, and `errorMessage2`.  Add some context `cimError` or something like that

**Review by spowelljr — CHANGES_REQUESTED** [review]

**Review by sharifelgamal — APPROVED** [review]

**Pull request merged**